File: Exercise/Hard/07. Nested Square Root Evaluator/Code/Nested Square Root Evaluator.java
import java.util.Scanner;

public class NestedSquareRootEvaluator {

    public static double evaluateExpression(String expression) {
        expression = expression.replace("√", "Math.sqrt");

        try {
            javax.script.ScriptEngineManager mgr = new javax.script.ScriptEngineManager();
            javax.script.ScriptEngine engine = mgr.getEngineByName("JavaScript");
            Object result = engine.eval(expression);
            return Double.parseDouble(result.toString());
        } 
        catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
            return Double.NaN;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
    
        String expression = scanner.nextLine();
        double result = evaluateExpression(expression);
        System.out.println(result);
    }
}

File: Exercise/Hard/07. Nested Square Root Evaluator/Code/Nested Square Root Evaluator.py
import math

def evaluate_expression(expression):
    expression = expression.replace('√', 'math.sqrt')
    try:
        return eval(expression)
    except Exception as e:
        print(f"Error: {e}")
        return None

def main():
    expression = input()
    result = evaluate_expression(expression)
    if result is not None:
        print(result)

if __name__ == "__main__":
    main()
